groupby.hpp

```diff
@@ -101,7 +102,7 @@ class iter::impl::GroupProducer {
 
     KeyGroupPair* operator->() {
       set_key_group_pair();
-      return current_key_group_pair_.get();
+      return current_key_group_pair_.operator->();
```

Comment:
It could. value() checks if value is stored and throws exception if it is not, so it is slightly different behaviour. Would you like me to make that change?